### PR TITLE
feat: add labs page

### LINKED ISSUE #3686: [Demand] Plugin infra: Tooltip entry

### PR DESCRIPTION
**Pull request by yanzhihong23**
image

also close #3686

### CODE REVIEW

File: packages/icons/plugins/RedPacket.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
+import { createIcon } from '../utils'
+import type { SvgIcon } from '@material-ui/core'
+
+export const RedPacketIcon: typeof SvgIcon = createIcon(
+    'RedPacket',
+    <>
+        <path
```
Comment (septs) [inline packages/icons/plugins/RedPacket.tsx:7]: plz use aka.pw/svgomg reduce size

File: packages/dashboard/src/locales/zh.json [hunk 2]
```diff
@@ -79,5 +80,27 @@
     "settings_email_hosting_service_title": "Email Hosting Service",
     "settings_email_hosting_service_desc": "Unauthorized",
     "settings_log_out_title": "Log Out",
-    "settings_log_out_desc": "Please make sure your data is backed up and log out"
+    "settings_log_out_desc": "Please make sure your data is backed up and log out",
+    "labs_file_service": "File Service",
```
Comment (septs) [inline packages/dashboard/src/locales/zh.json:84]: only provide english, other language no need copy

File: packages/icons/plugins/RedPacket.tsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
+import { createIcon } from '../utils'
```
Comment (nuanyang233) [inline packages/icons/plugins/RedPacket.tsx:1]: It looks like it conflicts with [this](https://github.com/DimensionDev/Maskbook/pull/3443/files#diff-f800213e606c38fc4120426314959b69e9a57231aa80d166a9ee6771f80c6bad)

File: packages/shared/src/hooks/useRemoteControlledDialog.ts [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+/**
+ * Use a dialog state controlled by remote
+ */
+export function useRemoteControlledDialog<T extends { open: boolean }>(
+    event: UnboundedRegistry<T>,
+    onUpdateByRemote?: (ev: T) => void,
+    tabType: 'self' | 'activated' = 'self',
+): Result<T> {
+    const [HOOK_ID] = useState(uuid()) // create a id for every hook
```
Comment (septs) [inline packages/shared/src/hooks/useRemoteControlledDialog.ts:25]: extract the line code to hook function? (`@masknet/shared`)

File: packages/maskbook/src/utils/native-rpc/types.ts [hunk 5]
```diff
@@ -9,6 +10,12 @@ export interface WebviewAPIs {
     /** Definition of LaunchPage see https://github.com/DimensionDev/Maskbook/blob/master/packages/maskbook/src/settings/types.ts */
     getSettings(key: 'launchPageSettings'): Promise<LaunchPage>
     getConnectedPersonas(): Promise<string>
+    app_isPluginEnabled(id: string): Promise<boolean>
```
Comment (Jack-Works) [inline packages/maskbook/src/utils/native-rpc/types.ts:13]: Please split this feature (mobile API) into a separate PR.

File: packages/icons/utils/ssr.tsx [hunk 6]
```diff
@@ -41,6 +42,9 @@ function render() {
                     <main id="general">
                         <IconPreview icons={General} title="General" />
                     </main>
+                    <main id="plugins">
+                        <IconPreview icons={Plugins} title="Glugins" />
```
Comment (Jack-Works) [inline packages/icons/utils/ssr.tsx:46]:
```suggestion
                        <IconPreview icons={Plugins} title="Plugins" />
```

File: packages/maskbook/src/utils/native-rpc/types.ts [hunk 7]
```diff
@@ -9,6 +10,12 @@ export interface WebviewAPIs {
...
+    app_setPluginStatus(id: string, enabled: boolean): void
+    setting_getNetworkTraderProvider(network: 'eth' | 'bsc' | 'polygon'): Promise<TradeProvider>
```
Comment (guanbinrui) [inline packages/maskbook/src/utils/native-rpc/types.ts:15]:
```suggestion
    setting_getNetworkTraderProvider(network: NetworkType): Promise<TradeProvider>
```

File: packages/maskbook/src/utils/native-rpc/types.ts [hunk 8]
```diff
@@ -9,6 +10,12 @@ export interface WebviewAPIs {
...
+    setting_setNetworkTraderProvider(network: 'eth' | 'bsc' | 'polygon', provider: TradeProvider): void
```
Comment (guanbinrui) [inline packages/maskbook/src/utils/native-rpc/types.ts:16]:
```suggestion
    setting_setNetworkTraderProvider(network: NetworkType, provider: TradeProvider): void
```

File: packages/maskbook/src/plugins/Trader/services.ts [hunk 9]
```diff
@@ -33,3 +38,21 @@ currentChainIdSettings.addListener((chainId: ChainId) => {
...
+            ethNetworkTradeProviderSettings.value = tradeProvier
+            break
+        case NetworkType.Binance:
+            bscNetworkTradeProviderSettings.value = tradeProvier
+            break
+        case NetworkType.Polygon:
+            polygonNetworkTradeProviderSettings.value = tradeProvier
+            break
+        default:
+            break
```
Comment (guanbinrui) [inline packages/maskbook/src/plugins/Trader/services.ts:56]:
```suggestion
            unreachable(networkType)
            break
```

File: packages/maskbook/src/plugins/Trader/trending/useCurrentTradeProvider.ts [hunk 10]
```diff
@@ -1,20 +1,28 @@
...
+    if (!networkType) return TradeProvider.UNISWAP
+    switch (networkType) {
+        case NetworkType.Ethereum:
+            return ethNetworkTradeProvider
+        case NetworkType.Polygon:
+            return polygonNetworkTradeProvider
+        case NetworkType.Binance:
+            return bscNetworkTradeProvider
+        default:
+            return TradeProvider.UNISWAP
```
Comment (guanbinrui) [inline packages/maskbook/src/plugins/Trader/trending/useCurrentTradeProvider.ts:26]:
```suggestion
            unreachable(networkType)
            return TradeProvider.UNISWAP
```

File: packages/maskbook/src/plugins/Trader/services.ts (hunk 9 above)
Comment (septs) [reply packages/maskbook/src/plugins/Trader/services.ts:56]: the switch why not as mapping

```
// definition
tradeProvideSettings: Record<NetworkType, (provider: TradeProvider) => void> = {
    /* .... */
}

// usage
tradeProvideSettings?.[networkType](tradeProvider)
```

File: packages/maskbook/src/plugins/Trader/services.ts [hunk 11]
```diff
@@ -3,26 +3,31 @@ export * from './apis'
...
 import { ChainId, getNetworkTypeFromChainId, NetworkType } from '@masknet/web3-shared'
 import { currentChainIdSettings } from '../Wallet/settings'
-import { currentTradeProviderSettings, currentDataProviderSettings } from './settings'
+import {
+    currentTradeProviderSettings,
+    currentDataProviderSettings,
+    ethNetworkTradeProviderSettings,
+    bscNetworkTradeProviderSettings,
+    polygonNetworkTradeProviderSettings,
+} from './settings'
```
Comment (septs) [inline packages/maskbook/src/plugins/Trader/services.ts:12]: may can as getter function

```ts
function getTradeProviderSettings(type: NetworkType): ProviderSettings | undefined
```

File: packages/maskbook/src/utils/native-rpc/Web.ts [hunk 12]
```diff
@@ -31,20 +31,27 @@ export const WebviewAPI: WebviewAPIs = {
...
     },
     app_setPluginStatus: async (id: string, enabled: boolean) => {
         await Services.Settings.setPluginStatus(id, enabled)
     },
     setting_getNetworkTraderProvider: async (network) => {
+        let provider
         switch (network) {
             case NetworkType.Ethereum:
-                return await Services.Settings.getEthNetworkTradeProvider()
+                provider = await Services.Settings.getEthNetworkTradeProvider()
```
Comment (septs) [inline packages/maskbook/src/utils/native-rpc/Web.ts:44]: why not early returns?

Comment (yanzhihong23) [reply packages/maskbook/src/utils/native-rpc/Web.ts:44]: eslint error: no-return-await

File: packages/dashboard/src/pages/Settings/components/DataSourceSetting.tsx [hunk 13]
```diff
@@ -12,10 +13,9 @@ export default function DataSourceSetting() {
 
     return (
         <SettingSelect value={source} onChange={handleChange}>
-            // TODO: use DataProvider
-            <MenuItem value={0}>CoinGecko</MenuItem>
-            <MenuItem value={1}>CoinMarketCap</MenuItem>
-            <MenuItem value={2}>Uniswap Info</MenuItem>
+            <MenuItem value={MarketTrendProvider.COIN_GECKO}>CoinGecko</MenuItem>
```
[the remark this answers is not in the document]
Comment (yanzhihong23) [reply packages/dashboard/src/pages/Settings/components/DataSourceSetting.tsx:16]: Data source setting will be removed from `Settings` page. Do in other PR.

File: packages/maskbook/src/utils/native-rpc/Web.ts (hunk 12 above)
Comment (septs) [reply packages/maskbook/src/utils/native-rpc/Web.ts:44]:
```suggestion
                return Services.Settings.getEthNetworkTradeProvider()
```

Comment (septs) [reply packages/maskbook/src/utils/native-rpc/Web.ts:44]: `return await ....` -> `return ...`

File: packages/maskbook/src/plugins/Trader/trending/useCurrentTradeProvider.ts (hunk 10 above)
Comment (Jack-Works) [reply packages/maskbook/src/plugins/Trader/trending/useCurrentTradeProvider.ts:26]: Resolved in #3847

File: packages/maskbook/src/components/InjectedComponents/ToolboxHint.tsx [hunk 14]
```diff
@@ -202,49 +202,51 @@ export function ToolboxHint(props: ToolboxHintProps) {
...
-                </MenuItem>
-            ) : null,
-            operatingSupportedChainMapping[ITO_Plugin.ID] ? (
-                <MenuItem onClick={onClaimAllDialogOpen} className={classes.menuItem}>
-                    <Image src={ToolIconURLs.claim.image} width={19} height={19} />
-                    <Typography className={classes.text}>{ToolIconURLs.claim.text}</Typography>
+        items
+            .filter((x) => x.hide !== true)
+            .map((desc) => (
+                <MenuItem onClick={desc.onClick} className={classes.menuItem}>
```
Comment (septs) [inline packages/maskbook/src/components/InjectedComponents/ToolboxHint.tsx:247]:
```suggestion
            .map((desc, key) => (
                <MenuItem key={key} onClick={desc.onClick} className={classes.menuItem}>
```

Comment (Jack-Works) [reply packages/maskbook/src/components/InjectedComponents/ToolboxHint.tsx:247]: `useMenu` hook will add keys

**Review by Jack-Works — APPROVED** [review]

**Review by Jack-Works — APPROVED** [review]

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**